--- Dockerfile
+++ Dockerfile_zsh
@@ -16,6 +16,7 @@
 ARG MPC_VERSION=1.3.1
 ARG MPFR_VERSION=4.2.2
 ARG PDCURSES_VERSION=3.9
+ARG ZSH_VERSION=5.9

 RUN apt-get update && apt-get install --yes --no-install-recommends \
   build-essential curl libgmp-dev libmpc-dev libmpfr-dev m4 p7zip-full
@@ -36,7 +37,8 @@
     https://frippery.org/files/busybox/busybox-w32-$BUSYBOX_VERSION.tgz \
     https://github.com/universal-ctags/ctags/archive/refs/tags/v$CTAGS_VERSION.tar.gz \
     https://downloads.sourceforge.net/project/mingw-w64/mingw-w64/mingw-w64-release/mingw-w64-v$MINGW_VERSION.tar.bz2 \
-    https://downloads.sourceforge.net/project/pdcurses/pdcurses/$PDCURSES_VERSION/PDCurses-$PDCURSES_VERSION.tar.gz
+    https://downloads.sourceforge.net/project/pdcurses/pdcurses/$PDCURSES_VERSION/PDCurses-$PDCURSES_VERSION.tar.gz \
+    https://downloads.sourceforge.net/projects/zsh/zsh/$ZSH_VERSION/zsh-$ZSH_VERSION.tar.xz
 COPY src/SHA256SUMS $PREFIX/src/
 RUN sha256sum -c $PREFIX/src/SHA256SUMS \
  && tar xJf 7z$Z7_VERSION-src.tar.xz --xform 's%^%7z/%' \
@@ -52,7 +54,8 @@
  && tar xJf mpfr-$MPFR_VERSION.tar.xz \
  && tar xzf make-$MAKE_VERSION.tar.gz \
  && tar xjf mingw-w64-v$MINGW_VERSION.tar.bz2 \
- && tar xzf PDCurses-$PDCURSES_VERSION.tar.gz
+ && tar xzf PDCurses-$PDCURSES_VERSION.tar.gz \
+ && tar xzf zsh-$ZSH_VERSION.tar.xz
 COPY src/w64devkit.c src/w64devkit.ico src/libmemory.c src/libchkstk.S \
      src/alias.c src/debugbreak.c src/pkg-config.c src/vc++filt.c \
      src/peports.c src/profile $PREFIX/src/
@@ -350,6 +353,20 @@
         CC=$ARCH-gcc AR=$ARCH-ar CFLAGS="-I.. -Os -DPDC_WIDE" pdcurses.a \
  && cp wincon/pdcurses.a /deps/lib/libcurses.a \
  && cp curses.h /deps/include
+
+WORKDIR /zsh-$ZSH_VERSION
+RUN /zsh-$ZSH_VERSION/configure \
+        --prefix=$PREFIX \
+        --host=$ARCH \
+        --disable-dynamic \
+        --enable-static \
+        --enable-multibyte \
+        --with-term-lib=ncursesw \
+        -without-tcsetpgrp \
+        CFLAGS="-Os -I/deps/include" \
+        LDFLAGS="-s -L/deps/lib" \
+ && make -j$(nproc) \
+ && make install

 WORKDIR /libiconv
 RUN /libiconv-$LIBICONV_VERSION/configure \
